packages/alfa-aria/src/feature.ts

```diff
@@ -633,23 +633,26 @@ Feature.register(
       // If the <th> is within a <table> with errors, it doesn't really have a role.
       if (tableModel.isErr()) return None;
 
-      const cell = Iterable.find(tableModel.get().cells, cell => cell.element.equals(element));
+      const cell = Iterable.find(tableModel.get().cells, (cell) =>
+        cell.element.equals(element)
+      );
       // If the current element is not a cell in the table, something weird happened and it doesn't have a role.
       if (cell.isNone()) return None;
 
       // This is not fully correct. If the header has no variant, its role should be computed as a <td>
       // @see https://www.w3.org/TR/html-aam-1.0/#html-element-role-mappings
       //     "th (is neither column header nor row header, and ancestor table element has table role)"
       // and "th (is neither column header nor row header, and ancestor table element has grid role)"
-      return  cell.get().variant.map(
-        variant => { switch (variant) {
+      return cell.get().variant.map((variant) => {
+        switch (variant) {
           case Header.Variant.Column:
-          case Header.Variant.ColGroup: return "ColumnHeader";
+          case Header.Variant.ColGroup:
+            return "columnheader";
           case Header.Variant.Row:
-          case Header.Variant.RowGroup: return "RowHeader"
-
-        }}
-      );
+          case Header.Variant.RowGroup:
+            return "rowheader";
```

Comment:
The fact that I had to correct the casing here is somewhat screaming for a better solution (we shouldn't be able to return a name here which has not been registered as a role in https://github.com/Siteimprove/alfa/blob/master/packages/alfa-aria/src/role/structure/row-header.ts (or elsewhere).

I kinda guess that we can't have a dynamic type constraint that the string we return here must have been defined via `register` first…
I guess the easiest way would be to maintain a list of role (i.e. an enumeration) and only pick from there (both for `register` and for this).

That would also allow stricter typing of predicates such as `hasRole` where it can be good to prevent a typo like `hasRole("buttton")` (3 "t") that would make the test always fail. `hasRole(Role.Name.Button)` would be safer.

What do you think?

Reply:
This is indeed a little unfortunate. I have an idea of how to implement this nicely but haven't gotten around to actually doing it yet. The key is incrementally building up some sort of container with a union of all the valid role names and then extracting the union from this container. The container would then serve as a global, strongly-typed registry of the valid roles.